assignments/002-unix-intro/unix-intro.sh

```diff
@@ -0,0 +1,1085 @@
+
+
+
+#### Ex. 48: Print the _first_ 15 lines of `sam/DPCh_plate1_A05_S5.sam` to _stdout_
+
+head -n 15 sam/DPCh_plate1_A05_S5.sam
+@HD     VN:1.3  SO:coordinate
+@SQ     SN:NC_037097.1  LN:96198142
+@SQ     SN:NC_037098.1  LN:57406636
+@SQ     SN:NC_037099.1  LN:81079776
+@SQ     SN:NC_037100.1  LN:74299616
+@SQ     SN:NC_037101.1  LN:93271540
+@SQ     SN:NC_037102.1  LN:77009353
+@SQ     SN:NC_037103.1  LN:86953105
+@SQ     SN:NC_037104.1  LN:73297397
+@SQ     SN:NC_037105.1  LN:90872356
+@SQ     SN:NC_037106.1  LN:60898746
+@SQ     SN:NC_037107.1  LN:48052637
+@SQ     SN:NC_037108.1  LN:77127610
+@SQ     SN:NC_037109.1  LN:75288599
+@SQ     SN:NC_037110.1  LN:55471129
+
+
+
+#### Ex. 49: Catenate the file `count_files.sh` to _stdout_
+
+cat count_files.sh
+# This looks fun
+ls -lR  * | awk 'NF>7' | wc | awk '{print "Estimated", $1, "files in the current directory"}'
+
+
+#### Ex. 50: Note that `count_files.sh` is a short shell script. Try to 
+####         execute it with `./count_files.sh`
+
+
+ ./count_files.sh
+Estimated 79 files in the current directory
+
+
+
+#### Ex. 51: List (long) `count_files.sh` to see what the permissions are,
+####         and then add execute (`x`) permissions to users and groups with
+####         the "add-or-subtract permissions syntax" of `chmod`, then long list it again
+
+(base) [c836820202@colostate.edu@login12 002-unix-intro]$ ls -l count_files.sh
+-rwxr-xr-x. 1 c836820202@colostate.edu c836820202grp@colostate.edu 112 Jan 23 10:43 count_files.sh
+chmod u+x,g+x count_files.sh
+ls -l count_files.sh
+
+
+#### Ex. 52: Use chmod to change permissions of the files in the `silly`
+####         directory to match their names. For example for the first 
+####         one you would use: `chmod 770 FILE-rwxrwx---`. Use the octal
+####         specifiers to `chmod`. Do this for all 5 files (one chmod command for each),
+####         and at the end do `ls -l silly` to verify they are correct.
+
+(base) [c836820202@colostate.edu@login12 002-unix-intro]$ find . -type d -name 'DIR*' -exec chmod 775 {} \;
+(base) [c836820202@colostate.edu@login12 002-unix-intro]$ find . -type f -name 'FILE-rw-r-----' -exec chmod 640 {} \;
+(base) [c836820202@colostate.edu@login12 002-unix-intro]$ find . -type f -name 'FILE-rw-rw-r--' -exec chmod 664 {} \;
+(base) [c836820202@colostate.edu@login12 002-unix-intro]$ find . -type f -name 'FILE-rwxr-----' -exec chmod 740 {} \;
+(base) [c836820202@colostate.edu@login12 002-unix-intro]$ find . -type f -name 'FILE-rwxrwxr--' -exec chmod 774 {} \;
```

Comment:
With all of these it looks like you find some code to change all files matching a `find` pattern.  It is good to know how to do that, too.

You could also do with the more simple:
```sh
chmod 640 FILE-rw-r-----
chmod 664 FILE-rw-rw-r--
...
```